typescript/sdk/src/deploy/verify/ZKSyncContractVerifier.ts

```diff
@@ -0,0 +1,333 @@
+    input: ContractVerificationInput,
+    verificationLogger: Logger,
+  ): Promise<void> {
+    const contractType: string = input.isProxy ? 'proxy' : 'implementation';
+
+    verificationLogger.debug(`📝 Verifying ${contractType}...`);
+
+    const data = this.getImplementationData(chain, input, verificationLogger);
+
+    try {
+      const verificationId: string = await this.submitForm(
+        chain,
+        verificationLogger,
+        data,
+      );
+
+      verificationLogger.trace(
+        { verificationId },
+        `Retrieved verificationId from verified ${contractType}.`,
+      );
+    } catch (error) {
+      verificationLogger.debug(
+        { error },
+        `Verification of ${contractType} failed`,
+      );
+      throw error;
+    }
+  }
+
+  /**
+   * @notice Verifies a contract on the specified chain
+   * @param chain The name of the chain where the contract is deployed
+   * @param input The contract verification input data
+   * @param logger An optional logger instance (defaults to the class logger)
+   */
+  public async verifyContract(
+    chain: ChainName,
+    input: ContractVerificationInput,
+    logger = this.logger,
+  ): Promise<void> {
+    const verificationLogger = logger.child({
+      chain,
+      name: input.name,
+      address: input.address,
+    });
+
+    const metadata = this.multiProvider.tryGetChainMetadata(chain);
+    const rpcUrl = metadata?.rpcUrls[0].http ?? '';
+    if (rpcUrl.includes('localhost') || rpcUrl.includes('127.0.0.1')) {
+      verificationLogger.debug('Skipping verification for local endpoints');
+      return;
+    }
+
+    const explorerApi = this.multiProvider.tryGetExplorerApi(chain);
+    if (!explorerApi) {
+      verificationLogger.debug('No explorer API set, skipping');
+      return;
+    }
+
+    if (!explorerApi.family) {
+      verificationLogger.debug(`No explorer family set, skipping`);
+      return;
+    }
+
+    if (explorerApi.family === ExplorerFamily.Other) {
+      verificationLogger.debug(`Unsupported explorer family, skipping`);
+      return;
+    }
+
+    if (input.address === ethers.constants.AddressZero) return;
+    if (Array.isArray(input.constructorArguments)) {
+      verificationLogger.debug(
+        'Constructor arguments in legacy format, skipping',
+      );
+      return;
+    }
+
+    await this.verify(chain, input, verificationLogger);
+  }
+
+  /**
+   * @notice Submits the verification form to the explorer API
+   * @param chain The name of the chain where the contract is deployed
+   * @param verificationLogger A logger instance for verification-specific logging
+   * @param options Additional options for the API request
+   * @returns The response from the explorer API
+   */
+  private async submitForm(
+    chain: ChainName,
+    verificationLogger: Logger,
+    options?: Record<string, any>,
+  ): Promise<any> {
+    const { apiUrl, family } = this.multiProvider.getExplorerApi(chain);
+
+    const url = new URL(apiUrl);
```

Comment:
For bouckscout they said, not yet supporting the verification yet on zksync, but for zksync etherscan, we added support, you just need to put `family:etherscan`, it would be handled by normal contract verifier.

Reply:
great thanks, can you leave this as a comment in the code as well please? 🙏

Reply:
Sure